File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/ProductsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductsRow } from "./ProductsRow";

export interface ProductsColumns {
    Id: Column<ProductsRow>;
    Name: Column<ProductsRow>;
    Description: Column<ProductsRow>;
    Price: Column<ProductsRow>;
    CategoryName: Column<ProductsRow>;
    Photo: Column<ProductsRow>;
    DateAdded: Column<ProductsRow>;
    Quantity: Column<ProductsRow>;
}

export class ProductsColumns extends ColumnsBase<ProductsRow> {
    static readonly columnsKey = 'ECommerceWebsite.Products';
    static readonly Fields = fieldsProxy<ProductsColumns>();
}

File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/Products/ProductsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Columns;

[ColumnsScript("ECommerceWebsite.Products")]
[BasedOnRow(typeof(ProductsRow), CheckNames = true)]
public class ProductsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string CategoryName { get; set; }
    public string Photo { get; set; }
    public DateTime DateAdded { get; set; }
    public int Quantity { get; set; }
}